Added option to manage bookmarks per buffer

leeor opened the pull request:
This allows for different tabs/buffers to have different bookmarks lists, instead of only a window-wide list that is either work dir based or truly global for the user.

All tests have passed.

MattesGroeger commented:
That looks great! Would you mind also updating the `doc/bookmarks.txt` file for completeness please?

Thanks in advance :+1:

leeor commented:
Happy you like it. It solved my issue with getting the plugin to work with gvim (getting the working dir set early enough).

Anyway, I've updated the bookrmarks.txt file and fixed the enable value in README.md.

MattesGroeger commented:
Great, much appreciated!
